freqtrade/optimize/hyperopt.py

```diff
@@ -609,10 +1050,267 @@ def load_previous_results(results_file: Path) -> List:
+                self.opt.void_loss = VOID_LOSS
+                self.opt.void = False
+                self.opt.rs = self.random_state
+            # in single mode restore the points directly to the optimizer
+            # but delete first in case we have filtered the starting list of points
+            self.opt = Hyperopt.opt_clear(self.opt)
+            rs = self.random_state
+            self.Xi[rs] = []
+            self.track_points()
+            if len(self.Xi[rs]) > 0:
+                self.opt.tell(self.Xi[rs], self.yi[rs], fit=False)
+            # delete points since in single mode the optimizer state sits in the main
+            # process and is not discarded
+            self.Xi, self.yi = {}, {}
+        del opts[:]
+
+    def setup_points(self):
+        self.n_initial_points, self.min_epochs, self.search_space_size = self.calc_epochs(
+            self.dimensions, self.n_jobs, self.effort, self.total_epochs, self.n_points
+        )
+        logger.info(f"Min epochs set to: {self.min_epochs}")
+        # reduce random points by n_points in multi mode because asks are per job
+        if self.multi:
+            self.opt_n_initial_points = self.n_initial_points // self.n_points
+        else:
+            self.opt_n_initial_points = self.n_initial_points
+        logger.info(f'Initial points: {self.n_initial_points}')
+        # if total epochs are not set, max_epoch takes its place
+        if self.total_epochs < 1:
+            self.max_epoch = int(self.min_epochs + len(self.trials))
+        # initialize average best occurrence
+        self.avg_best_occurrence = self.min_epochs // self.n_jobs
+
+    def return_results(self):
+        """
+        results are passed by queue in multi mode, or stored by ask_and_tell in single mode
+        """
+        batch_results = []
+        if self.multi:
+            while not backend.results_batch.empty():
+                worker_results = backend.results_batch.get()
+                batch_results.extend(worker_results)
+        else:
+            batch_results.extend(self.batch_results)
+            del self.batch_results[:]
+        return batch_results
+
+    def main_loop(self, jobs_scheduler):
+        """ main parallel loop """
+        try:
+            with parallel_backend('loky', inner_max_num_threads=2):
+                with Parallel(n_jobs=self.n_jobs, verbose=0, backend='loky') as parallel:
+                    jobs = parallel._effective_n_jobs()
+                    logger.info(f'Effective number of parallel workers used: {jobs}')
+                    # update epochs count
+                    opt_points = self.opt_points
+                    prev_batch = -1
+                    epochs_so_far = len(self.trials)
+                    epochs_limit = self.epochs_limit
+                    columns, _ = os.get_terminal_size()
+                    columns -= 1
+                    while epochs_so_far > prev_batch or epochs_so_far < self.min_epochs:
```

Comment:
Will need investigation.

`os.get_terminal_size()` breaks several tests - but i'm not certain it's still needed, as we're now printing hyperopt results via table.